src/content/docs/en/recipes/build-custom-img-component.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: Build a custom image component 
+description: Learn how to build a custom image component that supports media queries using the getImage function
+i18nReady: true
+type: recipe
+---
+
+Astro provides a couple of built in components that you can use to display and optimize your images.  The `<Picture>` component allows you to display responsive images and work with different formats and sizes. The `<Image>` componet will optimze your images and allow you to pass in different formats and quality properties.
+
+But there will be times where you will need to have more customizable options for your images that the `<Picture>` and `<Image>` components currently do not support. 
+
+In this recipe, you will learn how use the `getImage()` function to create your own custom image component that supports the use of media queries/art direction.  The purpose of the `getImage()` function is to create images for use outside of direct embedding in HTML.
+
+## Recipe
+
+1. Create a new Astro component and import the `getImage()` function
+
+    ```astro title="src/components/MyCustomImageComponent.astro"
+    ---
+     import { getImage } from "astro:assets";
+    ---
+
+    ```
+
+2. Destructure the `props` from `Astro.props`. You can also use TypeScript to type the props. 
+
+   ```astro title="src/components/MyCustomImageComponent.astro"
```

Comment:
```suggestion
   ```astro title="src/components/MyCustomImageComponent.astro" ins={11}
```
Up to you! Just making sure you're aware of our syntax highlighting options that we can show inserted lines as green by marking the line numbers like this. If there are only certain words or phrases you want to show inserted, you can do strings too, like `ins="getImage"`

Reply:
oh that is really cool. I will go ahead and add that in 👍